about/yocto-wpe.md

```diff
@@ -0,0 +1,112 @@
+
+
+### Combined instructions for Raspberry Pi 3B/3B+ (32-bit) and Raspberry Pi 4 (64-bit)
+
+Choose one of:
+```
+MACHINE=raspberrypi3-mesa
+```
+or:
+```
+MACHINE=raspberrypi4-64-mesa
+```
+
+Then to download the image:
+```
+SDIMG="core-image-weston-browsers-${MACHINE}.wic"
+SDCARD_DEV=/dev/mmcblk0_REMOVE_THIS
+wget https://wk-contrib.igalia.com/yocto/meta-webkit/browsers/stable/images/${MACHINE}/${SDIMG}.{bz2,bmap}
+```
+
+To write with '`bmaptool`':
+```
+sudo bmaptool copy "${SDIMG}".bz2 "${SDCARD_DEV}"
+```
+
+To write with '`dd`':
+```
+bzip2 -dc "${SDIMG}".bz2 | sudo dd of="${SDCARD_DEV}" bs=4k status=progress
+```
+
+## Set-up
+
+After writing the micro-SD card and using it to boot the device, it
+should boot to a graphical environment, using Weston (from Wayland),
+with a minimal top panel bar and some icons within it.
+
+To test WPE networking is needed, either by pluggin-in an ethernet
+cable (with DHCP networking should set-up automatically), or by
+clicking the "wireless" icon in the panel, that brings up a basic
+script to help to set the network up (selecting SSID and password).
+
+
+## Testing and demos
+
+After having the network working, the browser (`cog`) can be launched
+from one of the icons in the panel, or alternatively by opening a
+terminal with another of the icons and launching it from the command
+line.
+
+This is a quick list of environment variables and key bindings that
+can be used with `cog` to perform certain actions or change its
+behaviour (e.g. toggle fullscreen, reload, exit):
+* [https://github.com/Igalia/cog/blob/master/docs/platform-wl.md](https://github.com/Igalia/cog/blob/master/docs/platform-wl.md)
+
+Example:
+
+```
+COG_PLATFORM_WL_VIEW_FULLSCREEN=1 cog -P wl "https://www.wikipedia.org/"
+```
+
+
+### Demos
+
+This is a short list of suggested demos that can be used to check WPE
+in action on the device and its performance.
+
+| Name | Animation |
+|:----|:---:|
+| CSS transformation and animations in [WebKit poster circle example](https://webkit.org/blog-files/3d-transforms/poster-circle.html) | ![Image: WebKit poster circle demo](../assets/balena-wpe/postercircle.gif "WebKit poster circle demo") |
+| Playing an embedded video from [wpewebkit.org](https://wpewebkit.org/) | ![Image: Embedded video demo](../assets/balena-wpe/wpewebkitorg.gif "Embedded video demo") |
+| Playing [BouncyBalls](https://bouncyballs.org/) | ![Image: Playing BouncyBalls demo](../assets/balena-wpe/bouncyballs.gif "Playing BouncyBalls demo") |
+| Rendering WebGL [Aquarium](https://webglsamples.org/aquarium/aquarium.html) demo | ![Image: Rendering WebGL Aquarium demo](../assets/balena-wpe/webgl_aquarium.gif "Rendering WebGL Aquarium demo") |
+| Rendering WebGL [Abstract Shapes](https://mrdoob.neocities.org/023/) | ![Image: Rendering WebGL Abstract Shapes demo](../assets/balena-wpe/webgl_abstractshapes.gif "Rendering WebGL Abstract Shapes demo") |
```

Comment:
Would it make sense to duplicate the resources and put them under `../assets/yocto-images/`, or even to create new ones off the actual Yocto images instead of re-using the Balena ones?

Reply:
I think that recreating this would be busywork for nothing.  These are not "accurate representations of execution with Yocto images", like in a video, when for example you can see the framerate or if there was some effect/feature that could be seen with Yocto that couldn't be seen with the Balena one.

Basically the images were already there, and I thought that the page would be just more pretty with the pictures, I think that the page is better this way than only with text.

But I think that replicating the images with Yocto is not worth the effort or that we should stop for doing this right now, IMHO... maybe we can revisit this in the future if/when adding new/more demos or something.

Reply:
I understand what yo mean but still I think we should aim for having something better there than those animations anyway, they feel a bit too "hackish" to be in the official site of WPE to be honest.

For instance, I think we should aim for having proper screen recordings taken in the devices, just showing the desktop and Cog running inside of it. I don't think there's any real value on having a low quality video recorded with an external camera (a phone even?) showing the actual screen and the rpi running the WPE image beyond assuring the viewer that the image is running on the actual device... but even in that case the current gif animations are not showing the actual thing since they are showcasing the balena images anyway.

Anyway, I won't block the PR for this, but I think we should produce better images/videos (i.e. better quality, better framerate) if we are going to use them here, but that can be a follow up PR.

Reply:
> I don't think there's any real value on having a low quality video recorded with an external camera (a phone even?) showing the actual screen and the rpi running the WPE image beyond assuring the viewer that the image is running on the actual device...

But if we only capture the screen then we cannot see @psaavedra 's hand playing with BouncyBalls... :stuck_out_tongue_winking_eye: 

More seriously... I take the point of being a bit hackish and so on, with the camera moving etc, but OTOH the ones with the hand interacting show things than a simple screen capture wouldn't (also with the set-up of the device, screen, etc).  One can even appreciate the smoothness of scrolling with the old website, for example, and the immediacy of the stream starting playing when the finger presses.

It would be worth considering what we really want to show, to see if we use simple images (or nothing at all), or small videos, or what.

> [...] but even in that case the current gif animations are not showing the actual thing since they are showcasing the balena images anyway.

(FWIW: IIRC the binaries used in Balena are actually created in the same way, so it's basically the same thing)

Reply:
Like I said in the morning in a call, I think these demonstrations can be still improved, which is important if we think this website is product-oriented (vs developers-oriented), to properly showcase WPE.

Like I said, I don't see that the value of seeing the actual stuff (screen, board, fingers...) is more important than providing higher quality recordings of Cog performing on a RaspberryPi board, which for me it's more prioritary. Also, it's also arguable how smoothness can be appreciated on a low-FPS gif animation... from my POV, we should aim for having higher quality mini-demos here, and if we want to show the actual devices then let's do it properly with a decent recording setup: e.g. a good camera on a tripod, good framing of the scene, good lightning...

In any case, this is not a blocker, I'm Ok with merging this as-is now, but I think we should work on better mini-demos in a follow-up PR, that is on par with the kind of product-oriented quality we want  on wpewebkit.org.